rllib/core/rl_trainer/tests/test_trainer_runner.py

```diff
@@ -2,20 +2,81 @@
 import unittest
 
 import tensorflow as tf
+import torch
 import ray
+import time
 
+from typing import Type, Union
+
+from ray.rllib.core.rl_module import RLModule
+from ray.rllib.core.rl_trainer.rl_trainer import RLTrainer
 from ray.rllib.core.rl_trainer.trainer_runner import TrainerRunner
-from ray.rllib.core.testing.tf.bc_module import DiscreteBCTFModule
-from ray.rllib.core.testing.tf.bc_rl_trainer import BCTfRLTrainer
 from ray.rllib.policy.sample_batch import DEFAULT_POLICY_ID, MultiAgentBatch
 from ray.rllib.utils.test_utils import get_cartpole_dataset_reader
 
+Optimizer = Union[tf.keras.optimizers.Optimizer, torch.optim.Optimizer]
+
+
+def _get_trainer_class(framework: str) -> Type[RLTrainer]:
+    if framework == "tf":
+        from ray.rllib.core.testing.tf.bc_rl_trainer import BCTfRLTrainer
+
+        return BCTfRLTrainer
+    elif framework == "torch":
+        from ray.rllib.core.testing.torch.bc_rl_trainer import BCTorchRLTrainer
+
+        return BCTorchRLTrainer
+    else:
+        raise ValueError(f"Unsupported framework: {framework}")
+
+
+def _get_module_class(framework: str) -> Type[RLModule]:
+    if framework == "tf":
+        from ray.rllib.core.testing.tf.bc_module import DiscreteBCTFModule
+
+        return DiscreteBCTFModule
+    elif framework == "torch":
+        from ray.rllib.core.testing.torch.bc_module import DiscreteBCTorchModule
+
+        return DiscreteBCTorchModule
+    else:
+        raise ValueError(f"Unsupported framework: {framework}")
+
+
+def _get_optimizer_default_class(framework: str) -> Type[Optimizer]:
+    if framework == "tf":
+        return tf.keras.optimizers.Adam
+    elif framework == "torch":
+        return torch.optim.Adam
+    else:
+        raise ValueError(f"Unsupported framework: {framework}")
+
+
+def _get_trainer_runner(
```

Comment:
Remember that this is a unittest for TrainerRunner. The moment you introduce algorithm config it will become an integration test with algorithmConfig object. So I don't think any of these stuff apply to this particular file. If you look at [#31262](https://github.com/ray-project/ray/pull/31262) we do the stuff you mentioned for the top level integration. There are even unittests for what you just described.